bin/faucet/src/main.rs

```diff
@@ -99,30 +99,28 @@ async fn run_faucet_command(cli: Cli) -> anyhow::Result<()> {
             let account_file = AccountFile::read(&config.faucet_account_path)
                 .context("failed to load faucet account from file")?;
 
-            let faucet = Faucet::load(account_file, &mut rpc_client).await?;
+            let faucet =
+                Faucet::load(account_file, &mut rpc_client, config.remote_tx_prover_url).await?;
 
             // Maximum of 100 requests in-queue at once. Overflow is rejected for faster feedback.
             let (tx_requests, rx_requests) = mpsc::channel(100);
 
             let server =
                 Server::new(faucet.faucet_id(), config.asset_amount_options.clone(), tx_requests);
 
-            // Run both the client and the server concurrently.
-            // TODO: We probably want a more graceful shutdown.
-            let faucet =
-                async { faucet.run(rpc_client, rx_requests).await.context("faucet failed") };
-            let server =
-                async move { server.serve(config.endpoint).await.context("server failed") };
-            let mut tasks = tokio::task::JoinSet::new();
-            tasks.spawn(faucet);
-            tasks.spawn(server);
-            // SAFETY: join_next returns None if there are no tasks, and we definitely have tasks.
-            tasks
-                .join_next()
-                .await
-                .unwrap()
-                .context("failed to join task")?
-                .context("a server task ended unexpectedly")?;
+            // Run the server in a separate task
+            let server_handle = tokio::spawn(async move {
+                server.serve(config.endpoint).await.context("server failed")
+            });
+
+            // Run the faucet directly on the current thread
+            let faucet_result = faucet.run(rpc_client, rx_requests).await;
```

Comment:
I could be wrong but I think because this blocks here, `server_handle` is never awaited nor polled so I think it could error without us knowing. Maybe `server_handle` and `faucet.run()` could be polled with `tokio::select!`.

Reply:
I've just pushed a commit fixing this 👌🏼